crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -1966,6 +2010,7 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
                         // `@overload`ed functions without a body in unreachable code.
                         true
                     }
+                    Type::Dynamic(DynamicType::Divergent(_)) => true,
```

Comment:
I'm wondering if you have a better sense of why we need this. There is one test in `overloads.md` that fails without it, but in that that test `overload` resolves to `Never`, and I don't see where there would be a cycle at all. I'm wondering if you already debugged this and understand why we see a `Divergent` type here in that test?